// @condorhero/vuepress-plugin-export-pdf-v2/dist/shared/vuepress-plugin-export-pdf-v2.875ebc55.cjs
'use strict';

const vuepressPluginExportPdfCore = require('@condorhero/vuepress-plugin-export-pdf-core');
const node_path = require('node:path');
const node_module = require('node:module');
const vuepress = require('vuepress');
const debug = require('debug');

function _interopDefaultCompat (e) { return e && typeof e === 'object' && 'default' in e ? e.default : e; }

const debug__default = /*#__PURE__*/_interopDefaultCompat(debug);

const name = "@condorhero/vuepress-plugin-export-pdf-v2";
const type = "module";
const version$1 = "3.0.2";
const description = "VuePress 2.x plugin exports a PDF ðŸ“.";
const author = "CondorHero";
const license = "MIT";
const homepage = "https://github.com/condorheroblog/vuepress-plugin-export-pdf#readme";
const repository = {
	type: "git",
	url: "git+https://github.com/condorheroblog/vuepress-plugin-export-pdf.git",
	directory: "packages/vuepress-plugin-export-pdf-v2"
};
const bugs = {
	url: "https://github.com/condorheroblog/vuepress-plugin-export-pdf/issues"
};
const keywords = [
	"vuepress-plugin-export-pdf-v2",
	"vuepress-plugin-export-pdf",
	"vuepress-plugin",
	"vuepress",
	"plugin"
];
const exports$1 = {
	".": {
		types: "./dist/index.d.ts",
		require: "./dist/index.cjs",
		"import": "./dist/index.mjs"
	}
};
const main = "dist/index.cjs";
const module$1 = "dist/index.mjs";
const types = "dist/index.d.ts";
const typesVersions = {
	"*": {
		"*": [
			"./dist/*",
			"./dist/index.d.ts"
		]
	}
};
const bin = {
	"press-export-pdf": "bin/press-export-pdf.mjs"
};
const files = [
	"bin",
	"dist"
];
const engines = {
	node: ">=14"
};
const scripts = {
	build: "unbuild",
	prepublishOnly: "nr build",
	dev: "esno ./src/commands/press-export-pdf.ts export ./example/vuepress-next/docs"
};
const peerDependencies = {
	vuepress: ">=2.0.0-beta.49 <=2.x"
};
const dependencies = {
	"@condorhero/vuepress-plugin-export-pdf-core": "workspace:^",
	debug: "^4.3.4"
};
const devDependencies = {
	"@types/debug": "^4.1.8",
	vuepress: "^2.0.0-beta.61"
};
const publishConfig = {
	access: "public",
	registry: "https://registry.npmjs.org"
};
const pkg = {
	name: name,
	type: type,
	version: version$1,
	description: description,
	author: author,
	license: license,
	homepage: homepage,
	repository: repository,
	bugs: bugs,
	keywords: keywords,
	exports: exports$1,
	main: main,
	module: module$1,
	types: types,
	typesVersions: typesVersions,
	bin: bin,
	files: files,
	engines: engines,
	scripts: scripts,
	peerDependencies: peerDependencies,
	dependencies: dependencies,
	devDependencies: devDependencies,
	publishConfig: publishConfig
};

const devDebug = debug__default("vuepress-plugin-export-pdf-v2:dev-server");
const moduleRequire = node_module.createRequire((typeof document === 'undefined' ? require('u' + 'rl').pathToFileURL(__filename).href : (document.currentScript && document.currentScript.src || new URL('shared/vuepress-plugin-export-pdf-v2.875ebc55.cjs', document.baseURI).href)));
const { version } = moduleRequire("vuepress/package.json");
const serverApp = async (dir = "docs", commandOptions = {}) => {
  vuepressPluginExportPdfCore.checkEnv("VuePress", pkg.engines.node, version, pkg.peerDependencies.vuepress);
  const sourceDir = node_path.join(process.cwd(), dir);
  if (commandOptions.debug)
    debug__default.enabled("vuepress-plugin-export-pdf-v2:*");
  devDebug("sourceDir: %s", sourceDir);
  let userConfig = {};
  const userConfigPath = commandOptions.config ? vuepressPluginExportPdfCore.resolveUserConfigPath(commandOptions.config) : vuepressPluginExportPdfCore.resolveUserConfigConventionalPath(sourceDir);
  if (userConfigPath)
    userConfig = await vuepressPluginExportPdfCore.loadModule(userConfigPath);
  if (Array.isArray(userConfig.routePatterns))
    userConfig.routePatterns = ["/**", "!/404.html", ...userConfig.routePatterns];
  else
    userConfig.routePatterns = ["/**", "!/404.html"];
  const vuepressOutFile = commandOptions.outFile ?? `vuepress-${vuepressPluginExportPdfCore.timeTransformer()}.pdf`;
  const vuepressOutDir = commandOptions.outDir ?? ".";
  devDebug("userConfig: %O", userConfig);
  const {
    theme,
    bundler,
    sorter,
    puppeteerLaunchOptions,
    pdfOptions,
    routePatterns,
    outFile = vuepressOutFile,
    outDir = vuepressOutDir,
    urlOrigin = commandOptions.urlOrigin,
    pdfOutlines = commandOptions.pdfOutlines,
    outlineContainerSelector
  } = userConfig;
  const { userConfig: vuePressConfig } = await vuepress.loadUserConfig(vuepress.resolveUserConfigConventionalPath(sourceDir));
  const devApp = vuepress.createDevApp({
    source: sourceDir,
    ...vuePressConfig,
    bundler: bundler || vuePressConfig.bundler || vuepress.viteBundler(),
    theme: theme || vuePressConfig.theme || vuepress.defaultTheme(),
    host: "localhost",
    port: 8714
  });
  devApp.use(vuepress.transformUserConfigToPlugin(userConfig, sourceDir));
  await devApp.init();
  await devApp.prepare();
  const closeDevServer = await devApp.dev();
  process.stdout.write("Start to generate current site to PDF ...");
  const { pages, options: { temp: tempPath } } = devApp;
  const hashPages = pages.map((page) => ({
    // join => /docs//zh/
    path: node_path.join(`${devApp.siteData.base}${page.path}`),
    key: page.key
  }));
  try {
    await vuepressPluginExportPdfCore.generatePdf({
      pages: hashPages,
      tempDir: tempPath,
      port: devApp.options.port,
      host: devApp.options.host,
      outFile,
      outDir,
      sorter,
      urlOrigin,
      pdfOptions,
      pdfOutlines,
      routePatterns,
      puppeteerLaunchOptions,
      outlineContainerSelector
    });
  } catch (error) {
    console.error(error);
  }
  await closeDevServer();
  process.exit(0);
};

const registerCommands = (program) => {
  program.command("export [sourceDir]", "Export current VuePress site to a PDF file(default: docs)").allowUnknownOptions().option("-c, --config <config>", "Set path to config file").option("--outFile <outFile>", "Name of output file").option("--outDir <outDir>", "Directory of output files").option("--pdfOutlines <pdfOutlines>", "Keep PDF outlines/bookmarks").option("--urlOrigin <urlOrigin>", "Change the origin of the print url(Option displayHeaderFooter of pdfOptions is true)").option("--debug", "Enable debug mode").action(serverApp);
  program.command("info", "Display environment information").action(() => {
    vuepressPluginExportPdfCore.systemInfo([
      "@vuepress/bundler-vite",
      "@vuepress/bundler-webpack",
      "@vuepress/cli",
      "@vuepress/client",
      "@vuepress/core",
      "@vuepress/markdown",
      "@vuepress/plugin-active-header-links",
      "@vuepress/plugin-back-to-top",
      "@vuepress/plugin-container",
      "@vuepress/plugin-debug",
      "@vuepress/plugin-docsearch",
      "@vuepress/plugin-git",
      "@vuepress/plugin-google-analytics",
      "@vuepress/plugin-medium-zoom",
      "@vuepress/plugin-nprogress",
      "@vuepress/plugin-palette",
      "@vuepress/plugin-prismjs",
      "@vuepress/plugin-pwa",
      "@vuepress/plugin-pwa-popup",
      "@vuepress/plugin-register-components",
      "@vuepress/plugin-search",
      "@vuepress/plugin-shiki",
      "@vuepress/plugin-theme-data",
      "@vuepress/plugin-toc",
      "@vuepress/shared",
      "@vuepress/theme-default",
      "@vuepress/utils",
      "vuepress",
      "vuepress-vite",
      "vue",
      "vue-router",
      "vue-loader"
    ]);
  });
};
vuepressPluginExportPdfCore.runCli("press-export-pdf")(registerCommands);

exports.moduleRequire = moduleRequire;
exports.registerCommands = registerCommands;
exports.serverApp = serverApp;
